### ISSUE TITLE
Structured header parameter parser is too loose

### ISSUE THREAD

**Issue by gibson042** [tc39/ecmarkup#454]
A parameter name missing leading and/or trailing `_` is accepted, probably erroneously. For example, `_modifiers: a Modifiers,` (as [currently in proposal-regexp-modifiers spec](https://github.com/tc39/proposal-regexp-modifiers/blob/0736a543edf00de8487231688bfcbe89b74f8c54/spec.emu#L572)) renders like "_modifiers (a Modifiers)" with the literal `_` visible and "modifiers" not wrapped in a `<var>` element (as [currently in proposal-regexp-modifiers HTML](https://github.com/tc39/proposal-regexp-modifiers/blob/400bd3dc5ac9a1ab0837c863592fb0842f8a4bdc/index.html#L3364)).

The responsible code appears to be at https://github.com/tc39/ecmarkup/blob/994d663ee9245689f4d62baaffdcbfa055d5b404/src/header-parser.ts#L140

`/^[A-Za-z0-9_]+ */i` should probably be replaced with `/^_[A-Za-z0-9]+_ */i`, or maybe `/^(?:_[A-Za-z0-9]+_|[A-Za-z0-9]+) */i` if we want to support parameter names not wrapped in `_` (which I'd personally be against). And incidentally, I also note that the `i` flag is redundant with `A-Za-z`.

**Comment by bakkot** [comment]
Yeah, there should be a lint rule for this. (I prefer a lint rule to outright rejecting these inputs because it's nicer to get _some_ output while iterating.)
